_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -2302,6 +2302,30 @@ mfxStatus ImplementationAvc::AsyncRoutine(mfxBitstream * bs)
 
             if (newTask.m_picStruct[ENC] == MFX_PICSTRUCT_FIELD_BFF)
                 std::swap(newTask.m_type.top, newTask.m_type.bot);
+
+            mfxU64 timeStamp = newTask.m_timeStamp;
+            // preserve the time stamps.
+            if (m_timeStamps.size() && (timeStamp != static_cast<mfxU64>(MFX_TIMESTAMP_UNKNOWN)))
```

Comment:
Is it allowed to have UNKNOWN TS at input? It is inserted to the end and breaks the reordering. It should be computed somehow.

Reply:
Yes, the frame PTS can be UNKNOWN. 
If the input is UNKNOWN, it maybe make the timestamp shift a bit but doesn't break the reordering. The time stamps is still in increasing order except UNKOWN time stamp.
As the input is UNKNOWN, it is good enough to make output dts as UNKNOW, which is aligned with previous version behavior.